### ISSUE TITLE
Change to autoclosing of curly brackets

### ISSUE THREAD

**Issue by bland328** [mads-hartmann/bash-language-server#114]
_tl;dr: When typing `"${varname}"` in a bash script, the `{` doesn't autoclose because of the following `"`, but there's apparently a simple mechanism to change that, on a per-language bases._

If I'm understanding https://github.com/Microsoft/vscode/pull/52634#issuecomment-407207945 correctly _(if not, I apologize, being fairly new to VSCode)_, language server providers can now define which characters are allowed to:

> fall after the cursor as a prerequisite for [quote and bracket] autoclosing to occur.

Apparently, the default list of characters is `;:.,=}])> \n\t`.

I humbly propose that, for bash-language-server, double-quotes be added to that list, as a common idiom is to put double-quotes around variable references (e.g. `"${varname}"`).

In TextMate, it is lovely being able to type `"` (which autocloses), followed by `$`, followed by `{` (which also autocloses), and then `varname`.

In VSCode, the `{` doesn't autoclose because of the following `"`; which is why I propose adding it to the default `;:.,=}])> \n\t` list for bash-language-server.

I acknowledge there may be a downside I'm not considering :)

**Comment by skovhus** [comment]
Thank you for your contribution! @mads-hartmann any comments?

**Comment by mads-hartmann** [comment]
Hi @bland328, I think this would be a super useful change. These kind of rules are part of the language specification rather than the language server though, so in order to fix this, you'd have to head over to the [shellscript language definition](https://github.com/Microsoft/vscode/blob/master/extensions/shellscript/language-configuration.json) - which is part of vscode - and modify that accordingly 😊

I'm closing this issue, but I hope you'll take it up in the vscode repo 🙌
